docker/test/container_release_image_test.go

```diff
@@ -101,12 +127,67 @@ func TestCommitterReleaseImagesWithTLS(t *testing.T) {
 	}
 }
 
+// CreateAndStartSecuredDatabaseNode creates a containerized YugabyteDB or PostgreSQL
+// database instance in a secure mode.
+func startSecuredDatabaseNode(ctx context.Context, t *testing.T, params startNodeParameters) *dbtest.Connection {
+	t.Helper()
+
+	tlsConfig, credsPath := params.credsFactory.CreateServerCredentials(t, params.tlsMode, params.dbType, params.node)
+
+	node := &dbtest.DatabaseContainer{
+		DatabaseType: params.dbType,
+		Network:      params.networkName,
+		Hostname:     params.node,
+		TLSConfig:    tlsConfig,
+		CredsPathDir: credsPath,
+		UseTLS:       true,
+	}
+
+	node.StartContainer(ctx, t)
+	conn := node.GetConnectionOptions(ctx, t)
+
+	// This is relevant if a different CA was used to issue the DB's TLS certificates.
+	require.NotEmpty(t, node.TLSConfig.CACertPaths)
+	conn.TLS = connection.DatabaseTLSConfig{
+		Mode:       connection.OneSideTLSMode,
+		CACertPath: node.TLSConfig.CACertPaths[0],
+	}
+
+	// post start container tweaking
+	switch node.DatabaseType {
+	case testutils.YugaDBType:
+		// Ensure proper root ownership and permissions for the TLS certificate files.
+		node.ExecuteCommand(t, []string{
+			"chown", "root:root",
+			fmt.Sprintf("/creds/node.%s.crt", node.Hostname),
+			fmt.Sprintf("/creds/node.%s.key", node.Hostname),
+		})
+		node.EnsureNodeReadiness(t, "Data placement constraint successfully verified")
+		conn.Password = node.ReadPasswordFromContainer(t, containerPathForYugabytePassword)
+	case testutils.PostgresDBType:
+		// Ensure proper root ownership and permissions for the TLS certificate files.
+		node.ExecuteCommand(t, []string{
+			"chown", "postgres:postgres",
+			"/creds/server.crt",
+			"/creds/server.key",
+		})
+		node.EnsureNodeReadiness(t, dbtest.PostgresReadinessOutput)
+		node.ExecuteCommand(t, enforcePostgresSSLScript)
+		node.ExecuteCommand(t, reloadPostgresConfigScript)
+	default:
+		t.Fatalf("Unsupported database type: %s", node.DatabaseType)
+	}
+
+	t.Cleanup(
+		func() {
+			node.StopAndRemoveContainer(t)
+		})
```

Comment:
major: This should be right below the line that starts the container.
minor: style
```suggestion
	t.Cleanup(func() {
		node.StopAndRemoveContainer(t)
	})
```